### DISCUSSION TITLE
onnxruntime-web vs tfjs

### DISCUSSION THREAD

**Discussion by MariasStory** [vladmandic/human#323] [General]
**Issue Description**
tfjs models are slower then onnxruntime-web

**Steps to Reproduce**
https://github.com/microsoft/onnxruntime-web-benchmark

**Expected Behavior**
Use the fastest framework for a specific task

**Environment**
web

**Comment by vladmandic** [comment]
this is a veeeery generic statement without any provided information...  

- which exact models?
- using which backend?
- what is the performance difference - give specifics

**Comment by MariasStory** [comment]
Hi Vladimir,
Thanks for you comment. You are doing a really good job. I really like your solution.
I am sorry for not being specific.
In my test, I was using a simple image processing with models from https://github.com/PINTO0309/PINTO_model_zoo
There are different formats of the models, so it is easy to compare.
When using onnx models I have seen smaller memory footprint and faster execution.
There is a chance that it is a specifics of the models conversion, in my specific case.
I just want to suggest to try different approach. The onnx implementation is relatively easy to implement and there are plenty models to use.
In my case, I just used simplest single html implementation with:
`<script src="https://cdn.jsdelivr.net/npm/onnxruntime-web/dist/ort.min.js"></script>`

**Comment by vladmandic** [comment]
it really depends on the model and the backend used  
different models use different kernel ops and those ops are implemented differently by onnx and tjfs  

even more, they are very different comparing `tensorflow` backend (in `nodejs`) or `wasm` or `webgl` or `webgpu`.  
as far as i know:
- `onnx` really prefers `wasm` backend and support for `webgl` is quite limited
- `tfjs` doesn't spend that much time optimizing `wasm` backend, its the least one preferred and much more development is focused towards `webgl` (current) and `webgpu` (future)

that's why i'm asking - which exact model and which exact backend?

**Comment by vladmandic** [comment]
btw, as this is not really an issue, i'm converting it to a discussion item.
